### ISSUE TITLE
Add an "Add to Calendar" feature for each session

### ISSUE THREAD

**Issue by amberhinds** [wpa11yday/conference#69]
We received a request via email that I think is worth exploring. Can we add an "add to calendar" feature to each session so that people can easily add them to their calendars as reminders to attend?

This would be visible in two places:

1. On the session single floated right of the date/time in the header:
<img width="1200" alt="Screenshot 2024-09-10 at 11 00 13 AM" src="https://github.com/user-attachments/assets/78aea17f-3811-4a93-b60c-ae340bf86172">

2. On the schedule page in the view details accordion after the excerpt on the left side:
<img width="1175" alt="Screenshot 2024-09-10 at 11 02 56 AM" src="https://github.com/user-attachments/assets/984a65cf-bc40-4a7a-a724-d58c02e58d05">

Not sure if @joedolson has some code for this that could be modified from My Calendar?

**Comment by joedolson** [comment]
I do; but it's limited to just Google Calendar and an iCal download. We previously looked at this idea using a service, but the accessibility (as I recall) was dubious. I think the question is about exactly what mechanisms we want to offer. Each calendar service works differently, so we can't just have *one* link.

**Comment by amberhinds** [comment]
Yes, I did it with [Add to Calendar](https://addtocalendar.com/) on the [Save the Date blog post](https://2024.wpaccessibility.day/save-the-date-for-wordpress-accessibility-day-2024/). 

I had to modify the HTML that I copy/pasted from the Add to Calendar website to make it accessible, but it's fine. That might be workable again if we had a metabox where I could paste the HTML in to display on the desired spots on the front end, but it's a ton of work to do manually for each session each year since we have to set up each session on the Add to Calendar website first. I'm also not certain what their free plan limits are.

**Comment by joedolson** [comment]
Glad you could remember the details...but yeah, that's going to be too labor intensive to be worthwhile. 

If we choose to only support link-targetable calendars, then that's Gmail, Office 365, and Outlook.com; (also AOL and Yahoo, but...nah.)

We could do Add to Calendar as a disclosure that displays those three options. We could also provide a downloadable ICS, which makes it possible to add to any calendar, but is a pain to deal with for users.

**Comment by amberhinds** [comment]
I'm going to reach out to some people and see if anyone might be willing to contribute to this feature request.

**Comment by Camwyn** [comment]
I can help with this. We have similar functionality on TEC. I can do this over the next few days.

@amberhinds: 
Do you want a structure much like the one you linked to on the blog post? Or would you prefer something more like a drop-down?

Are there specific ones you want added (ical, gcal, etc) I'd assume you also want to add a downloaded-able ics as an option.

**Comment by amberhinds** [comment]
I think a drop-down functionality like you would have in The events
Calendar would be the cleanest. Is it possible to have all the same options
you have in TEC? Also, .ics would be great.

Thank you, Stephen!!

**Comment by amberhinds** [comment]
Any luck on this @Camwyn?

**Comment by Camwyn** [comment]
I've started working on it. I'm planning on doing more this weekend so we've got something Monday morning

**Comment by Camwyn** [comment]
Just for an update: a nasty flu ran through my household and I was out for the first half of the week. I just got back to semi-productive yesterday. I'm setting aside time today to wrap this up.

Sorry for the delay

**Comment by joedolson** [comment]
Closed in https://github.com/wpa11yday/conference/commit/4daee7c7e3545a56e7e15f1b6da16f33bf33e5b2
